lib/generators/rolemodel/optics/icons/templates/app/helpers/icon_helper.rb

```diff
@@ -1,78 +1,61 @@
-
-    tag.span(name, **options)
+  # filled, size, weight, emphasis, additional_classes, color, hover_text
+  def material_icon(name, **)
+    IconBuilder.from_library(:material, name, **).build
   end
 
-  def icon(name, filled: false, size: 'medium', weight: 'normal', emphasis: 'normal', color: nil, classes: nil, hover_text: name)
-    using_custom_icon = custom_icon_path(name).present?
-
-    contents = using_custom_icon ? embedded_svg("icons/#{name}.svg") : name
-
-    options = {
-      class: classes(using_custom_icon, filled, size, weight, emphasis, classes),
-      title: hover_text
-    }
-
-    if color
-      options[:style] = "#{using_custom_icon ? 'fill' : 'color'}: var(--rms-colors-#{color}-base);"
-    end
-
-    tag.span(contents, **options)
+  # duotone, filled, size, weight, additional_classes, color, hover_text
+  def phosphor_icon(name, **)
+    IconBuilder.from_library(:phosphor, name, **).build
   end
 
-  # Inspired by https://blog.cloud66.com/using-svgs-in-a-rails-stack
-  def embedded_svg(filename, options = {})
-    asset = Rails.application.assets_manifest.find_sources(filename).first
-
-    if asset
-      file = asset.source.force_encoding('UTF-8')
-      doc = Nokogiri::HTML::DocumentFragment.parse file
-      svg = doc.at_css 'svg'
-      svg['class'] = options[:class] if options[:class].present?
-    else
-      doc = "<!-- SVG #{filename} not found -->"
-    end
-
-    # These SVG files can safely be marked html_safe since we created them and they are part of this app's code.
-    raw doc # rubocop:disable Rails/OutputSafety
+  # filled, size, additional_classes, color, hover_text
+  def tabler_icon(name, **)
+    IconBuilder.from_library(:tabler, name, **).build
   end
 
-  private
-
-  def classes(using_custom_icon, filled, size, weight, emphasis, additional_classes = nil)
-    shape_class = filled ? 'icon--filled' : 'icon--outlined' # true, false
-    size_class = "icon--#{size}" # normal, large, x-large
-    weight_class = "icon--weight-#{weight}" # light, normal, semi-bold, bold
-    emphasis_class = "icon--#{emphasis}-emphasis" # low, normal, high
+  # size, additional_classes, color, hover_text
+  def feather_icon(name, **)
+    IconBuilder.from_library(:feather, name, **).build
+  end
 
-    base_class = using_custom_icon ? 'custom-icons' : 'material-symbols-outlined'
-    "#{base_class} #{shape_class} #{size_class} #{weight_class} #{emphasis_class} #{additional_classes}"
+  # size, additional_classes, color, hover_text
+  def lucide_icon(name, **)
+    IconBuilder.from_library(:lucide, name, **).build
   end
-  # rubocop:enable Metrics/ParameterLists
 
-  def custom_icon_path(name)
-    Rails.application.assets_manifest.find_sources("icons/#{name}.svg").first
+  # filled, size, weight, emphasis, additional_classes, color, hover_text
+  def custom_icon(name, **)
+    IconBuilder.from_library(:custom, name, **).build
   end
+
+  # Set the default to Material Icons
+  alias icon material_icon
```

Comment:
Might be more interesting to look into https://www.rubydoc.info/gems/thor/Thor/Shell/Basic#ask-instance_method

I suspect it would be a little more time consuming, but one could prompt which icon library they'd like the generator to "install" and inject this line dynamically based on what's selected (and also not bring in `lucide` or `feather` builders if they're not being used, for instance).

Reply:
Future improvement :) I'll add a comment.

There is no harm in having all the builders available to you even if you aren't using them. Makes it easier to swap out later in the project too so you don't have to go to RMS rails and pull the builder. Maybe could have them commented out or something but the failure mode is pretty obvious.